internal/cluster/cluster_service.go

```diff
@@ -284,11 +312,54 @@ func (m *membersMap) reset() {
 
 func (m *membersMap) logMembers(version int32, members []pubcluster.MemberInfo) {
 	// synchronized in Update
-	sb := strings.Builder{}
-	sb.WriteString(fmt.Sprintf("\n\nMembers {size:%d, ver:%d} [\n", len(m.members), version))
-	for _, mem := range members {
-		sb.WriteString(fmt.Sprintf("\tMember %s - %s\n", mem.Address, mem.UUID))
+	m.logger.Info(func() string {
+		sb := strings.Builder{}
+		sb.WriteString(fmt.Sprintf("\n\nMembers {size:%d, ver:%d} [\n", len(m.members), version))
+		for _, mem := range members {
+			sb.WriteString(fmt.Sprintf("\tMember %s - %s\n", mem.Address, mem.UUID))
+		}
+		sb.WriteString("]\n\n")
+		return sb.String()
+	})
+}
+
+// LargerGroupMajorMinorVersion finds the version of the most numerous member group by version.
+func LargerGroupMajorMinorVersion(members []pubcluster.MemberInfo) (uint16, error) {
+	// synchronized in Update
+	// The members should have at most 2 different version (ignoring the patch version).
+	var vs [2]uint16
+	var vv [2]pubcluster.MemberVersion
+	var count [2]int
+	var next int
+	majority := len(members) / 2
+	for i, mem := range members {
+		if mem.LiteMember {
+			continue
+		}
+		v := mem.Version.MajorMinor()
+		if v == vs[0] {
+			count[0]++
+		} else if v == vs[1] {
+			count[1]++
+		} else if next < 2 {
+			vs[next] = v
+			vv[next] = mem.Version
+			count[next]++
+			next++
+		} else {
+			return 0, fmt.Errorf("more than 2 distinct member versions found: %s, %s, %s", vv[0], vv[1], mem.Version)
+		}
+		if i > majority {
+			break
+		}
```

Comment:
Is this logic correct? I mean, we should break only if one of the groups is larger than the majority.

Reply:
`i` corresponds to `currentCount` in the Java side, so I think that code may be also wrong. And in the Java side that piece of code only runs when there is the `localMember` (i.e., never) so I'll just remove it.

Reply:
Removed at: https://github.com/hazelcast/hazelcast-go-client/pull/695/commits/67c7887a6db6650f61b79de879d8b6456be75afe